### DISCUSSION TITLE
Disappearing clock icon.

### DISCUSSION THREAD

**Discussion by SteveRMann** [HASwitchPlate/HASPone#232] [Troubleshooting]
I have a sensor near my trash bin in the garage that sets a Home Assistant helper entity to true if the bin is in the garage, false if not.
The plan is on the evening before the trash pickup, I will blink the colors of the clock button.

I have two Node Red functions that toggle between the button colors.
(Each function node is toggled by a once per second timer node.)

```
//----- Desktop Black FG -----
msg.topic = "hasp/desktop/command/p[1].b[7].pco";
msg.payload = 0;
node.send(msg);

//----- Desktop Yellow BG -----
msg.topic = "hasp/desktop/command/p[1].b[7].bco";
msg.payload = 65504;
node.send(msg);

return null;

```

```
//----- Desktop White FG -----
msg.topic = "hasp/desktop/command/p[1].b[7].pco";
msg.payload = 65535;
node.send(msg);

//----- Desktop Black BG -----
msg.topic = "hasp/desktop/command/p[1].b[7].bco";
msg.payload = 0;
node.send(msg);

return null;
```

This all works just fine.  When the bin is present, the clock button blinks between white text on a black background, and black text on a yellow background.

My question is: Why does the clock icon disappear?

**Comment by aderusha** [comment]
If you're looking to combine this with the existing Clock + Icon blueprint, the problem you have is that any of the blueprints that use icons are using some hacky approaches to make those icons happen.  Nextion doesn't let us do things like combine different two different text fields with different justification in the same field, so you can't directly have a button with arbitrary text on one side with an arbitrary icon on the other.

The blueprints I made handle this by way of sending the text, pausing some msec, then sending the icon via the nextion `xstr` command which will park itself anywhere on screen on top of what might already be there.  [See this line for an example](https://github.com/HASwitchPlate/HASPone/blob/main/Home_Assistant/blueprints/hasp_Display_Clock_with_Icon.yaml#L336).

The problem is that an icon positioned via `xstr` will be overwritten by anything happening to the object underneath it.  So when the blueprint puts the clock text up, then sends the clock icon, it's fine.  When node-red comes behind it and changes the background color, that overwrites the icon placed in front of the button.

I'm just spitballing here, but maybe log traffic to get the `xstr` string coming from the blueprint, and then add that to your node-red automation?  That way it'll paint the clock, paint the icon, then node red will change the color which removes the icon, then you paint it back on there again.  It's a hack on top of a hack but maybe might work.

**Reply by SteveRMann** [reply]
Got it.  I happened to be watching when the time incremented a minute and the icon was there until I changed the fg/bg colors.  My solution, the, is to simply use another button, like the Garage Light.  No icon.
